spec/controllers/article_controller_spec.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ArticlesController, type: :controller do
+  describe 'GET #index' do
+    subject { get :index }
+    include_examples 'not_authorized'  
+
+    context 'when login' do
+      let(:articles) { create_list(:article, 4) }
+      let(:user) { create(:user) }
+      before { sign_in user }
```

Comment:
as we can see that we perform `sign_in` in every `describe` box, we may refactor it this way
```ruby
# spec/controllers/article_controller_spec.rb

RSpec.describe ArticlesController, type: :controller do
  let(:user) { create(:user) }

  before { sign_in user }

  describe 'GET #index' do
    subject { get :index }

    include_examples 'not_authorized'

    let(:articles) { create_list(:article, 4) }

    it 'return 200' do
      expect(subject.status).to eq(200)
    end

    it 'return data' do
      subject
      expect(assigns(:articles)).to match_array(articles)
    end
  end

  # the rest ...
end
```
and
```ruby
# spec/support/share_example/not_authorized.rb

shared_examples 'not_authorized' do
  let(:user) { create(:user) }

  context 'when user is not authorized' do
    before { sign_out user }

    it 'return 302' do
      subject
      expect(subject.status).to eq(302)
    end
  end
end
```